v2/pkg/genruntime/resource_reference.go

```diff
@@ -111,6 +109,13 @@ func (ref ResourceReference) Validate() error {
 	return nil
 }
 
+func (ref ResourceReference) ToNamespacedRef(namespace string) NamespacedResourceReference {
```

Comment:
Maybe add a test for this? I guess it's covered by other things though.

Reply:
Good prodding to add test, found some awkward behavior (not really a bug but confusing). Clarified it, added comment, and added test.